### PR TITLE
Replace RoutingSettings with PersistenceExtensions  for settings

### PR DESCRIPTION
**Pull request by SeanFeldman**
Change PartitionAwareConfiguration to work with `PersistenceExtensions<CosmosDbPersistence>` for settings rather than `RoutingSettings<TransportDefinition>`.

### CODE REVIEW

File: src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/SynchronizedStorage.cs [hunk 1]
```diff
@@ -8,13 +8,23 @@ class SynchronizedStorage : Feature
...
+            }
+
+            var databaseName = context.Settings.Get<string>(SettingsKeys.DatabaseName);
+
             var partitionConfig = context.Settings.Get<PartitionAwareConfiguration>();
 
-            context.Container.ConfigureComponent(() => new StorageSessionFactory(databaseName, clientFactory(), partitionConfig), DependencyLifecycle.SingleInstance);
+            if (partitionConfig is null)
+            {
+                throw new Exception("No message partition mappings were found. Use transport.Partition() to configure mappings.");
```
Comment (danielmarbach) [inline src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/SynchronizedStorage.cs:24]: This should say persistence.Partition but I can fix that later

**Review by danielmarbach — APPROVED** [review]
Originally I put it under routing extensibility because I felt it is kind of a routing concern. I can see though how it can also be seen as more of a persistence concern

File: src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/SynchronizedStorage.cs (hunk 1 above)
Comment (SeanFeldman) [reply src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/SynchronizedStorage.cs]: 👍 done

**Comment by danielmarbach** [comment]
Cherry picked over while resolving merge conflicts locally

**Pull request closed without merging**